dot/sync/chain_sync.go

```diff
@@ -166,26 +175,45 @@ func newChainSync(cfg chainSyncConfig) *chainSync {
 		workerPool:         newSyncWorkerPool(cfg.net, cfg.requestMaker),
 		badBlocks:          cfg.badBlocks,
 		requestMaker:       cfg.requestMaker,
+		waitPeersDuration:  cfg.waitPeersDuration,
 	}
 }
 
+func (cs *chainSync) waitEnoughPeersAndTarget() <-chan struct{} {
+	resultCh := make(chan struct{})
+
+	go func() {
+		defer close(resultCh)
+		for {
+			cs.workerPool.useConnectedPeers()
+			_, err := cs.getTarget()
+
+			totalAvailable := cs.workerPool.totalWorkers()
+			if totalAvailable >= uint(cs.minPeers) && err == nil {
+				return
+			}
+
+			time.Sleep(cs.waitPeersDuration)
```

Comment:
I think we should listen on `cs.stopCh` here and have a `select` here to exit early if we need to.